crowbar_framework/lib/crowbar/deployment_queue.rb

```diff
@@ -19,9 +19,9 @@ class DeploymentQueue
 
     attr_reader :logger, :proposal_queue
 
-    def initialize(logger: Rails.logger, queue: ProposalQueue.new)
+    def initialize(logger: Rails.logger)
       @logger         = logger
-      @proposal_queue = queue
+      @proposal_queue = ::ProposalQueue.first
```

Comment:
the order is always correct if we use the first record, as this is always the oldest